CONTRIBUTION.md

```diff
@@ -24,17 +22,162 @@ These commands can be executed from the project root:
+	handler := wait.New(func() (waitFinished bool, response *foo.GetBarResponse, err error) {
+		s, err := a.GetBarExecute(ctx, BarId, projectId)
+		if err != nil {
+			return false, nil, err
+		}
+		if s.Id == nil || s.Status == nil {
+			return false, nil, fmt.Errorf("could not get Bar id or status from response for project %s and Bar %s", projectId, BarId)
+		}
+		if *s.Id == BarId && (*s.Status == UpdateSuccess) {
+			return true, s, nil
+		}
+		if *s.Id == BarId && (*s.Status == UpdateFail) {
+			return true, s, fmt.Errorf("update failed for Bar with id %s", BarId)
+		}
+		return false, nil, nil
+	})
+	handler.SetTimeout(30 * time.Minute)
+	return handler
+}
+
+// DeleteBarWaitHandler will wait for Bar deletion
+func DeleteBarWaitHandler(ctx context.Context, a APIClientInterface, BarId, projectId string) *wait.AsyncActionHandler[foo.GetBarResponse] {
+	handler := wait.New(func() (waitFinished bool, response *foo.GetBarResponse, err error) {
+		s, err := a.GetBarExecute(ctx, BarId, projectId)
+		if err != nil {
+			return false, nil, err
+		}
+		if s.Id == nil || s.Status == nil {
+			return false, nil, fmt.Errorf("could not get Bar id or status from response for project %s and Bar %s", projectId, BarId)
+		}
+		if *s.Id == BarId && *s.Status == DeleteSuccess {
+			return true, s, nil
+		}
+		if *s.Id == BarId && *s.Status == DeleteFail {
+			return true, s, fmt.Errorf("delete failed for Bar with id %s", BarId)
+		}
+		return false, nil, nil
+	})
+	handler.SetTimeout(20 * time.Minute)
+	return handler
+}
+```
+
+#### Notes
+
+- The states may vary from service to service
+- The `id` and the `state` might not be present on the root level of the API response, this also varies from service to service. You must always match the resource `id` and the resource `state` to what is expected
+- The timeout values included above are just for reference, each resource takes different amounts of time to finish the create, update or delete operations
+- For some resources, after a successful delete operation the resource can't be found anymore, so a call to the `Get` method would result in an error. In those cases, the waiter can be implemented by calling the `List` method and check that the resource is not present, like in this example:
+
+  ```go
+  // DeleteBarWaitHandler will wait for Bar deletion
+  func DeleteBarWaitHandler(ctx context.Context, a APIClientInterface, barId, projectId string) *wait.AsyncActionHandler[foo.ListBarsResponse] {
+      handler := wait.New(func() (waitFinished bool, response *foo.ListBarsResponse, err error) {
+          s, err := a.ListBarsExecute(ctx, barId, projectId)
+          if err != nil {
+              return false, nil, err
+          }
+          for i := range s {
+              if *s[i].Id == barId {
+                  return false, nil, nil
+              }
+          }
+          return true, s, nil
+      })
+      handler.SetTimeout(10 * time.Minute)
+      return handler
+  }
+  ```
+
```

Comment:
I was thinking and we could actually replace those cases with a call to the `List` endpoint, not sure why we did it against the `404` error in the first place.

But for the guidelines I would not suggest that approach because it's much more error-prone

Reply:
Agreed, that approach isn't the greatest